zqd/handlers_zeek_test.go

```diff
@@ -16,98 +16,179 @@ import (
+		assert.InDelta(t, 1437, info.Size, 2)
+		assert.True(t, info.PacketSupport)
+		assert.Equal(t, p.pcapfile, info.PacketPath)
+		// XXX Cannot test with full struct because zeek's randomly generate
+		// uids can vary by 1 characater in size. Should probably be tested with
+		// the same seed set in zeek but for now just test individual fields.
+		// expected := api.SpaceInfo{
+		// Name:          p.space,
+		// MaxTime:       &max,
+		// MinTime:       &min,
+		// Size:          1437,
+		// PacketSupport: true,
+		// PacketSize:    0, // XXX This is not supported right now. Fix this.
+		// PacketPath:    p.pcapfile,
+		// }
+		// assert.Equal(t, expected, info)
+	})
+	t.Run("PacketIndexExists", func(t *testing.T) {
+		require.FileExists(t, filepath.Join(p.core.Root, p.space, packet.IndexFile))
+	})
+	t.Run("ResponseMessages", func(t *testing.T) {
+		info, err := os.Stat(p.pcapfile)
+		require.NoError(t, err)
+		require.Len(t, p.payloads, 3)
+		status := p.payloads[1].(*api.PacketPostStatus)
+		assert.Equal(t, status.Type, "PacketPostStatus")
+		assert.Equal(t, status.PacketSize, info.Size())
+		assert.Equal(t, status.PacketReadSize, info.Size())
+	})
 }
 
-func (s *PacketPostSuite) TestNoPacketFilter() {
-	expected := ""
-	res := execSearch(s.T(), s.core, s.space, "_path = packet_filter | count()")
-	s.Equal(expected, res)
+func TestPacketPostInvalidPcap(t *testing.T) {
+	p := packetPost(t, zeekpath, "./testdata/invalid.pcap", 500)
+	defer p.cleanup()
+	t.Run("ErrorMessage", func(t *testing.T) {
+		// XXX Better error message here.
+		require.Equal(t, "Unknown magic 73696874\n", string(p.body))
+	})
 }
 
-func (s *PacketPostSuite) TestSpaceInfo() {
-	u := fmt.Sprintf("http://localhost:9867/space/%s", s.space)
-	res := httpSuccess(s.T(), zqd.NewHandler(s.core), "GET", u, nil)
-	var info api.SpaceInfo
-	err := json.NewDecoder(res).Decode(&info)
-	s.NoError(err)
-	s.Equal(s.pcapfile, info.PacketPath)
-	s.True(info.PacketSupport)
-	s.Equal(nano.Ts(1501770877471635000), *info.MinTime)
-	s.Equal(nano.Ts(1501770880988247000), *info.MaxTime)
+func TestPacketPostZeekFailImmediate(t *testing.T) {
+	exec := filepath.Join(cwd, "testdata", "zeekstartfail.sh")
+	p := packetPost(t, exec, "./testdata/valid.pcap", 202)
+	defer p.cleanup()
+	t.Run("TaskEndError", func(t *testing.T) {
+		expected := &api.TaskEnd{
+			Type: "TaskEnd",
+			// XXX This is dependent on execution order. TaskID is global when
+			// should be attached to instance of core.
+			TaskID: 2,
+			Error: &api.Error{
+				Type: "Error",
+				// XXX This is not an informative failure message. Will fix in
+				// followup pr.
+				Message: "exit status 2",
+			},
+		}
+		require.Contains(t, p.payloads, expected)
+	})
 }
 
-func (s *PacketPostSuite) TestWritesIndexFile() {
-	stat, err := os.Stat(filepath.Join(s.core.Root, s.space, packet.IndexFile))
-	s.NoError(err)
-	s.NotNil(stat)
+func TestPacketPostZeekFailAfterWrite(t *testing.T) {
+	exec := filepath.Join(cwd, "testdata", "zeekwritefail.sh")
```

Comment:
Was there something you wanted to verify related to the logs that `zeekwritefail.sh` wrote?

Reply:
No there was not. It was more relevant when the api was waiting on zeek logfiles to return status code 2xx... I still think it's an interesting test and it also ensures that no data remains in the space (with the subsequent test that calls to spaceInfo).